dev/com.ibm.ws.app.manager.module/src/com/ibm/ws/app/manager/module/internal/SimpleDeployedAppInfoBase.java

```diff
@@ -276,10 +316,14 @@ public Container getContainer() {
                     }
                 };
                 this.classesContainerInfo.add(containerInfo);
-                Set<String> resolved = new HashSet<String>();
-                ManifestClassPathHelper.addCompleteJarEntryUrls(this.classesContainerInfo, entry, jarContainer, resolved);
+
+                List<ContainerInfo> manifestClassPaths = new ArrayList<>();
+                ManifestClassPathHelper.addCompleteJarEntryUrls(manifestClassPaths, entry, jarContainer, new HashSet<String>());
+                manifestClassPathConsumer.consume(manifestClassPaths, this.classesContainerInfo);
+
+                // NOTE: tWAS does not do recursive search like this
```

Comment:
It can be done pretty easily by adding a method to ManifestClassPathConsumer to indicate if recursive searching should be done, but I decided not to because this is such a strange corner case that I cannot imaging a tWAS user having an application with such nested JARs and they expect them not to be added to the class path.